napari/utils/progress.py

```diff
@@ -57,6 +60,8 @@ class progress(tqdm):
     """
 
     monitor_interval = 0  # set to 0 to disable the thread
+    progress_list = EventedList()
```

Comment:
yes, this!   this along with `progress.progress_list.events.inserted.connect(self.make_new_pbar)` above is exactly what I had in mind.

Maybe make this `EventedSet` instead? (not sure if these are hashable ... but they could be)

I like this pattern a lot.  any gotchas i'm not immediately seeing?